### ISSUE TITLE
Add ability to provide url to a recipe website and extract the recipe information

### ISSUE THREAD

**Issue by kylec32** [kylec32/recipe-holder#4]
Take the ingredient import to the next level by just being able to provide a URL to recipe website and it auto fill the information. (Even if it just handled one popular recipe website that would be cool).

**Comment by sschadwick** [comment]
Perhaps this would be easier to do via a recipe API rather than scraping as website? Interested in helping either way, simply looking for direction.

**Comment by kylec32** [comment]
Yeah I have no problem with that. Haven't really looked into recipe APIs. As long as it's just as easy for the user I'm all for it. Structured data definitely sounds better.

**Comment by tylerwalters** [comment]
Are there any free recipe APIs that include popular food websites and blogs that is also available for free? I did a quick search for some recipe APIs, and found a couple that include a lot of the popular sites, but they require a monthly subscription, which I assume is not feasible for an open source project like this. 

Seems like this could be handled by scraping data from the provided urls. I think the best solution would be to scrape urls for [Schema.org recipe data](https://schema.org/Recipe), and similar structured data, because it would add support for a large number of websites. Then support could be added to any popular recipe websites that are not covered with the structured data. 

I have some experience with scraping data from sites, and with Schema.org metadata, if you would like me to take a crack at it.
